### PR TITLE
feat: added Homepage

### PR DESCRIPTION
**Pull request by srv-pranjal**
# Added Home Page
## Description

 - Added Navbar and Routes
 - Added Backend for Categories
 - Added Category Component
 - Added Hero Component
 
homepage

### CODE REVIEW

File: src/Components/Category/Category.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import { useState, useEffect } from "react";
+import axios from "axios";
+import { CategoryCard } from "./CategoryCard";
+
+export const Category = () => {
+  const [category, setCategory] = useState([]);
+  useEffect(() => {
```
Comment (Rishi05051997) [inline src/Components/Category/Category.js:8]: Error handling is missing here if in some cases categories not able to fetch then you need to show something you can do this in future

File: src/Components/Category/CategoryCard.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import { Link } from "react-router-dom";
+
+export const CategoryCard = ({ categoryName, image, altText }) => {
```
Comment (Rishi05051997) [inline src/Components/Category/CategoryCard.js:4]: Very Good you are using separate category card component

**Review by Rishi05051997 — COMMENTED** [review]
Good to go

File: src/Components/Category/Category.js (hunk 1 above)
Comment (srv-pranjal) [reply src/Components/Category/Category.js:8]: Yes, I missed that. Thanks

File: src/App.js [hunk 3]
```diff
@@ -1,41 +1,16 @@
...
-          >
-            Contribute
-          </a>
-        </div>
-      </header>
+      <Navbar />
+      <Routes>
+        <Route path="/" element={<Home />} />
+        <Route path="/mock" element={<Mockman />} />
+      </Routes>
```
Comment (maitrakhatri) [inline src/App.js:13]: add route for 404 page

File: src/Components/Category/Category.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    (async () => {
+      const response = await axios.get("/api/categories");
+      setCategory(response.data.categories);
+    })();
+  }, []);
```
Comment (maitrakhatri) [inline src/Components/Category/Category.js]: do api calls in try catch blocks for error handlining

File: src/assets/index.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+export const hero = require("./hero.jpg");
+export const headphone = require("./headphone.jpg");
+export const laptop = require("./laptop.jpg");
+export const mobile = require("./mobile.jpg");
+export const smartwatch = require("./smartwatch.jpg");
+export const speaker = require("./speaker.jpg");
+export const storage = require("./storage.jpg");
```
Comment (maitrakhatri) [inline src/assets/index.js:7]: you can use image source paths directly just like we do in HTML, for that your assets should be in `public` folder. 
Converting every single image in component with take lot of code and time

**Review by maitrakhatri — APPROVED** [review]
Good folder structure
Code is easy to read and understand
left some suggestions

File: src/Components/Category/Category.js (hunk 4 above)
Comment (srv-pranjal) [reply src/Components/Category/Category.js]: Fixed, Thanks.

File: src/assets/index.js (hunk 5 above)
Comment (srv-pranjal) [reply src/assets/index.js:7]: Okay, I was doing like this for exporting all images from a single file.

File: src/App.js (hunk 3 above)
Comment (srv-pranjal) [reply src/App.js:13]: Sure, I will

**Pull request merged**